node/pkg/p2p/registry.go

```diff
@@ -18,12 +18,17 @@ type registry struct {
 	// Per-chain error counters
 	errorCounters  map[vaa.ChainID]uint64
 	errorCounterMu sync.Mutex
+
+	// Per-chain last VAA timestamp (Unix nanoseconds)
+	lastVaaTimestamps   map[vaa.ChainID]int64
+	lastVaaTimestampsMu sync.Mutex
```

Comment:
```suggestion
	lastVaaTimestampsMu sync.RWMutex
```
You could consider using the RWMutex here instead. That way `GetLastVaaTimestamp` only needs to acquire a ReadLock, and `SetLastVaaTimestamp` could acquire a write lock. This should allow for concurrent reads of the LastVAATimestamp.

Not sure if this would have a meaningful impact on performance, but I do think it makes sense anyway.

Reply:
Very sensible